code/modules/mob/living/simple_animal/bot/mulebot.dm

```diff
@@ -1,866 +1,810 @@
+			usr << browse(null,"window=mulebot")
+
+
+
+// returns true if the bot has power
+/mob/living/simple_animal/bot/mulebot/proc/has_power()
+	return !open && cell && cell.charge > 0 && wires.HasPower()
+
+/mob/living/simple_animal/bot/mulebot/proc/buzz(type)
+	switch(type)
+		if(SIGH)
+			audible_message("[src] makes a sighing buzz.", "<span class='italics'>You hear an electronic buzzing sound.</span>")
+			playsound(loc, 'sound/machines/buzz-sigh.ogg', 50, 0)
+		if(ANNOYED)
+			audible_message("[src] makes an annoyed buzzing sound.", "<span class='italics'>You hear an electronic buzzing sound.</span>")
+			playsound(loc, 'sound/machines/buzz-two.ogg', 50, 0)
+		if(DELIGHT)
+			audible_message("[src] makes a delighted ping!", "<span class='italics'>You hear a ping.</span>")
+			playsound(loc, 'sound/machines/ping.ogg', 50, 0)
+
+
+// mousedrop a crate to load the bot
+// can load anything if emagged
+/mob/living/simple_animal/bot/mulebot/MouseDrop_T(atom/movable/AM, mob/user)
+
+	if(user.incapacitated() || user.lying)
+		return
+
+	if (!istype(AM))
+		return
+
+	load(AM)
+
+// called to load a crate
+/mob/living/simple_animal/bot/mulebot/proc/load(atom/movable/AM)
+	if(load ||  AM.anchored)
+		return
+
+	if(!isturf(AM.loc)) //To prevent the loading from stuff from someone's inventory or screen icons.
+		return
+
+	var/obj/structure/closet/crate/CRATE
+	if(istype(AM,/obj/structure/closet/crate))
+		CRATE = AM
+	else
+		if(wires.LoadCheck())
+			buzz(SIGH)
+			return		// if not emagged, only allow crates to be loaded
+
+	if(CRATE) // if it's a crate, close before loading
+		CRATE.close()
+
+	if(isobj(AM))
+		var/obj/O = AM
+		if(O.buckled_mob || (locate(/mob) in AM)) //can't load non crates objects with mobs buckled to it or inside it.
+			buzz(SIGH)
+			return
+
+	if(isliving(AM))
+		if(!load_mob(AM))
+			return
+	else
+		AM.loc = src
+
+	load = AM
+	mode = BOT_IDLE
+	update_icon()
+
+/mob/living/simple_animal/bot/mulebot/proc/load_mob(mob/living/M)
+	if(M.buckled)
+		return 0
+	/*
+	var/turf/T = get_turf(src)
+	if(M.loc != T)
+		density = 0
+		var/can_step = step_towards(M, T)
+		density = 1
+		if(!can_step)
+			return 0
+	*/
+	if(M.client)
+		M.client.perspective = EYE_PERSPECTIVE
+		M.client.eye = src
+	passenger = M
+	load = M
+	M.loc = src
+	return 1
+
+
+// called to unload the bot
+// argument is optional direction to unload
+// if zero, unload at bot's location
+/mob/living/simple_animal/bot/mulebot/proc/unload(dirn)
+	if(!load)
+		return
+
+	mode = BOT_IDLE
+
+	overlays.Cut()
+
+	load.loc = loc
+	load.pixel_y = initial(load.pixel_y)
+	load.layer = initial(load.layer)
+	if(dirn)
+		var/turf/T = loc
+		var/turf/newT = get_step(T,dirn)
+		if(load.CanPass(load,newT)) //Can't get off onto anything that wouldn't let you pass normally
+			step(load, dirn)
+	if(passenger)
+		if(passenger.client)
+			passenger.client.eye = passenger.client.mob
+			passenger.client.perspective = MOB_PERSPECTIVE
+		passenger = null
+	load = null
```

Comment:
So you can't get on top of the mulebot anymore, you're inside of it? It's not consistent with how crates are loaded.